devito/passes/clusters/aliases.py

```diff
@@ -818,12 +831,40 @@ def lower_schedule(cluster, schedule, chosen, sregistry, options):
     return clusters, subs
 
 
-def rebuild(cluster, others, schedule, subs):
+def pick_best(variants):
+    """
+    Use the variant score and heuristics to return the variant with the best
+    trade-off between operation count reduction and working set increase.
+    """
+    best = variants.pop(0)
+    for i in variants:
+        best_flop_score, best_ws_score = best.score
+        if best_flop_score == 0:
+            best = i
+            continue
+
+        i_flop_score, i_ws_score = i.score
+
+        # The current heustic is fairly basic: the one with smaller working
+        # set size increase wins, unless there's a massive reduction in operation
+        # count in the other one
+        delta = i_ws_score - best_ws_score
+        if (delta > 0 and i_flop_score / best_flop_score > 100) or \
```

Comment:
100? Shouldn't it bit some MIN_COST too?

Reply:
I have a subsequent (large) PR which changes lots of things around here, so I'll definitely do that but can it wait until the next PR?

Reply:
Fine with me